src/main/java/net/fabricmc/stitch/representation/JarClassEntry.java

```diff
@@ -153,20 +186,29 @@ public boolean isInterface() {
         return Access.isInterface(getAccess());
     }
 
+    public boolean hasEnclosingClass() {
+        return enclosingClass != null;
+    }
+
+    public boolean hasEnclosingMethod() {
+        return enclosingMethod != null;
+    }
+
     public boolean isAnonymous() {
-        return getName().matches("[0-9]+");
+        return hasEnclosingClass() && innerName == null;
+    }
+
+    public boolean isInner() {
+        return hasEnclosingClass() && !hasEnclosingMethod() && innerName != null;
```

Comment:
This will return true for this `Local`:
```java
class TopLevel {
  static { class Local {} }
}
```
and since this method isn't used and stitch is not an api, I recommend removing this method.